connectors/utils.py

```diff
@@ -401,3 +401,26 @@ async def func_to_execute(*args, **kwargs):
 
 def e2str(entry):
     return entry.name.lower()
+
+
+class MultiService:
+    def __init__(self, services):
+        self._services = services
+
+    async def run(self):
+        tasks = [asyncio.create_task(service.run()) for service in self._services]
+
+        done, pending = await asyncio.wait(tasks, return_when=asyncio.FIRST_COMPLETED)
+
+        for task in pending:
+            task.cancel()
+            await task
+
+    async def shutdown(self, sig):
+        logger.info(f"Caught {sig}. Graceful shutdown.")
+        for service in self._services:
+            logger.debug(f"Shutting down {service.__class__.__name__}...")
```

Comment:
`BaseService` has  a `name` attribute now

Reply:
Hmmm I cannot find it, can you point to the code that defines it?

Reply:
oh crap, my mistake, I was thinking about the DataSource classes :)

Reply:
@tarekziade I guess you confuse `BaseService ` with `BaseDataSource`

Reply:
yes! (we could add `name` to the service too I suppose)